python/pyspark/context.py

```diff
@@ -1010,13 +1010,42 @@ def setJobGroup(self, groupId, description, interruptOnCancel=False):
         ensure that the tasks are actually stopped in a timely manner, but is off by default due
         to HDFS-1208, where HDFS may respond to Thread.interrupt() by marking nodes as dead.
         """
+        warnings.warn(
```

Comment:
@squito, if I should write a warning, I think it will be like this. I don't feel strongly about this so I am not sure.

FYI, in case Python, with default setting, the warning will be issued once.

Reply:
Unless you feel strongly about it, I'd rather have a hard failure if you try to call this, not in pinned mode.

Did you intentionally want a warning even if you are in pinned mode, just to include the info about inherited threads?  I think we should take the down to info.  Also the docstring should be updated to include the caveat on inherited threads

Reply:
What I am worries are .. 

Firstly, people use it although it's buggy because it kind of works okay in single thread without a pin-thread mode. Seems like there is still possibility that another thread is launched and local properties are reset though.

Secondly, even with pin-thread mode, it does not work properly about inherited threads, yes, as you said.

About warning vs info, PySpark currently does not have a proper logging system .. so we should rely on manual printing out or warning. If we use manual printing way, it's tricky for users to control it. In case of warning, they can control, for instance, if they want to print out the warning only for the initial call or every call.

Reply:
Ah, I can maybe throw an error when the current thread is different from main thread when pin-thread mode is off, and warn when pin-thread mode is on maybe?

Reply:
ok, I see your point.  I don't feel great about any of our options here, but perhaps you are right.  I do wish there was a way we could warn more loudly when using this in non-pinned mode.

I'm just brainstorming here -- another thing we could do is just have the spark UI display some big warning for all of these fields whenever you're using pyspark in non-pinned mode (if we can reliabaly detect this).  Eg. job group would just say "Unavailable with pyspark in non-pinned mode -- see docs (link)".

(Even if we want to do that, doesn't need to be done here.)

Reply:
I don't know yet because this job group things (with non-pinned mode) work kind of fine in a single thread (without multiple threads). We might have to detect if the current thread is child thread (by checking if current thread is the main thread) .. but I think it will miss some corner cases ..